python/fusion_engine_client/applications/p1_filter.py

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env -S python3 -u
+
+from datetime import datetime
+import os
+import sys
+import time
+
+# Since stdout is used for data stream, don't write any print statements to stdout.
+# Done here to avoid any log/print statements triggered by imports.
+original_stdout = sys.stdout
+sys.stdout = sys.stderr
+
+# Add the Python root directory (fusion-engine-client/python/) to the import search path to enable FusionEngine imports
+# if this application is being run directly out of the repository and is not installed as a pip package.
+root_dir = os.path.normpath(os.path.join(os.path.dirname(__file__), '..'))
+sys.path.insert(0, root_dir)
+
+from fusion_engine_client.messages import MessagePayload, message_type_by_name
+from fusion_engine_client.parsers import FusionEngineDecoder
+from fusion_engine_client.utils.argument_parser import ArgumentParser, ExtendedBooleanAction
+
+
+if __name__ == "__main__":
+    parser = ArgumentParser(description="""\
+Filter FusionEngine data coming through stdin. Examples:
+  netcat 192.168.1.138 30210 | ./p1_filter.py --blacklist -m SatelliteInfo --display > /tmp/out.p1log
+  cat /tmp/out.p1log | ./p1_filter.py -m Pose > /tmp/pose_out.p1log
+  stty -F /dev/ttyUSB0 speed 460800 cs8 -cstopb -parenb -icrnl -ixon -ixoff -opost -isig -icanon -echo && cat /dev/ttyUSB0 | ./p1_filter.py -m Pose > /tmp/pose_out.p1log
+""")
+
+    parser.add_argument(
+        '-m', '--message-type', type=str, action='append',
+        help="An list of class names corresponding with the message types to whitelist (default) or blacklist. May be specified "
+             "multiple times (-m Pose -m PoseAux), or as a comma-separated list (-m Pose,PoseAux). All matches are"
+             "case-insensitive.\n"
+             "\n"
+             "If a partial name is specified, the best match will be returned. Use the wildcard '*' to match multiple "
+             "message types.\n"
+             "\n"
+             "Supported types:\n%s" % '\n'.join(['- %s' % c for c in message_type_by_name.keys()]))
+    parser.add_argument(
+        '--blacklist', action=ExtendedBooleanAction,
+        help="By default '--message-type' is a whitelist. If 'blacklist' is set, use them as a blacklist instead.")
+    parser.add_argument(
+        '--display', action=ExtendedBooleanAction,
+        help="Periodically display messages received and forwarded on sterr.")
```

Comment:
sterr --> stderr?

This sentence is a little ambiguous. I think you're saying that you will display the content on the stderr, but it could be read as "the messages that were forwarded on stderr".

Reply:
Also I don't think it's right? Down below you just print periodic status, you don't actually display messages at all.